test/FunctionalTests/Microsoft.OData.Client.Tests/Metadata/ClientTypeUtilTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ClientTypeUtilTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using Microsoft.OData.Client.Metadata;
+using System;
+using Xunit;
+
+namespace Microsoft.OData.Client.Tests.Metadata
+{
+    /// <summary>
+    /// A class test to test whether the use of Key Attribute on a Type property can identify a type as an entity when the conventional key naming methods are not used.
+    /// </summary>
+    public class ClientTypeUtilTests

```

Comment:
Test should include the method to see if the client will still read the key property and give out the right value for it. In this case there should be a test to get if it gets the Key as PId and would love to see a test that tests for a key name which is non conventional as the Attribute should sort this out.